### PR TITLE
MhvSecondaryNav feature-toggle/render bugfix

### PR DESCRIPTION
**Pull request by radavis**
## Summary

- Fixes bug where `<MhvSecondaryNav />` renders before loading feature toggles, so that navigation doesn't send users to unreleased pages.
- Adds some extra specs around the Medical Records link in the `<MhvSecondaryNav />` in order to clarify feature toggle behavior
- `<MhvSecondaryNav />` holds vertical space via `vads-u-visibility--hidden` class when loading.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#98927

## Testing done

- Adds unit specs to existing code.
- Local testing with mocker-api and delay

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [-] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNav.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    it('when mhvIntegrationMedicalRecordsToPhase1 enabled: /my-health/medical-records', () => {
+      const initialState = stateFn({
+        mhvIntegrationMedicalRecordsToPhase1: true,
+      });
+      const { getByRole } = setup({ initialState });
+      const mrLink = getByRole('link', { name: /^Records/ });
+      expect(mrLink.href).to.match(/my-health\/medical-records$/);
+    });
+  });
+});
```
Comment (radavis) [inline src/platform/mhv/secondary-nav/tests/MhvSecondaryNav.unit.spec.jsx:70]: Screenshot from 2024-12-10 16-45-56

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNavMRPage.unit.spec.jsx [hunk 2]
```diff
@@ -1,92 +0,0 @@
...
-
-    it('feature toggles disabled', () => {
-      const store = mockStore({
-        mhvIntegrationMedicalRecordsToPhase1: false,
-        mhvTransitionalMedicalRecordsLandingPage: false,
-      });
-      testForLink(store);
-    });
-  });
-});
```
Comment (radavis) [inline src/platform/mhv/secondary-nav/tests/MhvSecondaryNavMRPage.unit.spec.jsx:92]: Output of deleted spec:

Screenshot from 2024-12-11 13-22-58

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNav.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  it('renders', () => {
+    const initialState = stateFn({ loading: false });
+    const { getByRole } = setup({ initialState });
+    getByRole('navigation', { name: 'My HealtheVet' });
+    const mrLink = getByRole('link', { name: /^My HealtheVet/ });
+    expect(mrLink.href).to.match(/my-health$/);
+  });
+
+  describe('Medical Records href', () => {
```
Comment (radavis) [inline src/platform/mhv/secondary-nav/tests/MhvSecondaryNav.unit.spec.jsx:47]: I considered writing this section as below, but wasn't certain if it was easier to read and comprehend vs what's in the PR.

```js
    [
      {
        nextState: { mhvIntegrationMedicalRecordsToPhase1: true },
        href: '/my-health/medical-records',
      },
      {
        nextState: {
          mhvIntegrationMedicalRecordsToPhase1: true,
          mhvTransitionalMedicalRecordsLandingPage: true,
        },
        href: '/my-health/medical-records',
      },
      {
        nextState: { mhvTransitionalMedicalRecordsLandingPage: true },
        href: '/my-health/records',
      },
      {
        nextState: {
          mhvIntegrationMedicalRecordsToPhase1: false,
          mhvTransitionalMedicalRecordsLandingPage: false,
        },
        href: '/my-health/medical-records',
      },
    ].forEach(({ nextState, href }) => {
      const value = JSON.stringify(nextState);
      it(`when state is ${value}\n\t href is ${href}`, () => {
        const { getByRole } = setup({ initialState: stateFn(nextState) });
        const mrLink = getByRole('link', { name: /^Records/ });
        expect(mrLink.href).to.match(new RegExp(`${href}$`));
      });
    });
```

File: src/platform/mhv/secondary-nav/containers/MhvSecondaryNav.jsx [hunk 4]
```diff
@@ -57,10 +58,10 @@ export const mhvSecNavItems = [
 const MhvSecondaryNav = () => {
   const items = [...mhvSecNavItems];
   const {
-    loading,
+    loading = true,
```
Comment (nichia) [inline src/platform/mhv/secondary-nav/containers/MhvSecondaryNav.jsx:61]: Nice

**Review by nichia — DISMISSED** [review]
Nice and clear test spec.

**Review by nichia — DISMISSED** [review]
LGTM

**Review by dcloud — APPROVED** [review]
🚢

**Pull request merged**